hugegraph-common/src/main/java/com/baidu/hugegraph/config/HugeConfig.java

```diff
@@ -98,30 +70,33 @@ private void setLayoutIfNeeded(Configuration conf) {
         this.setLayout(propConf.getLayout());
     }
 
-    private static PropertiesConfiguration loadConfigFile(String path) {
+    private static Configuration loadConfigFile(String path) {
         E.checkNotNull(path, "config path");
         E.checkArgument(!path.isEmpty(),
                         "The config path can't be empty");
 
         File file = new File(path);
-        E.checkArgument(file.exists() && file.isFile() && file.canRead(),
-                        "Need to specify a readable config, but got: %s",
-                        file.toString());
-
-        PropertiesConfiguration config = new PropertiesConfiguration();
-        config.setDelimiterParsingDisabled(true);
-        try {
-            config.load(file);
-        } catch (ConfigurationException e) {
-            throw new ConfigException("Unable to load config: %s", e, path);
-        }
-        return config;
+        return loadConfigFile(file);
     }
 
     @SuppressWarnings("unchecked")
     public <T, R> R get(TypedOption<T, R> option) {
         Object value = this.getProperty(option.name());
-        return value != null ? (R) value : option.defaultValue();
+        if (value == null) {
+            return option.defaultValue();
+        }
+
+
+        Class dataType = option.dataType();
+        if (dataType.isInstance(value)) {
+            return (R) value;
+        }
+
+        if (value instanceof String) {
+            return option.parseConvert((String) value);
+        }
```

Comment:
don't call isInstance and isInstance as possible for every get()

Reply:
The config2 is not compatible， so, the get method should cover type, do you have any advise for this code impl?

Reply:
does the this.append(config) trigger addPropertyDirect()? if yes, we don't need to parse again